core/src/main/scala/dagr/core/execsystem/TaskManager.scala

```diff
@@ -518,17 +521,27 @@ class TaskManager(taskManagerResources: SystemResources = TaskManagerDefaults.de
     logger.debug("runSchedulerOnce: found " + readyTasks.size + " readyTasks tasks")
 
     // get the list of tasks to schedule
-    val tasksToSchedule: Map[UnitTask, ResourceSet] = scheduler.schedule(
-      runningTasks = runningTasks,
-      readyTasks   = readyTasks,
-      systemCores  = taskManagerResources.cores,
-      systemMemory = taskManagerResources.systemMemory,
-      jvmMemory    = taskManagerResources.jvmMemory
-    )
-    logger.debug("runSchedulerOnce: scheduling " + tasksToSchedule.size + " tasks")
-
-    // add the given tasks to the task runner with the appropriate (as determined by the scheduler) resources.
-    scheduleAndRunTasks(tasksToSchedule = tasksToSchedule)
+    val tasksToSchedule: Map[UnitTask, ResourceSet] = {
+      if (canDoAnything) {
```

Comment:
It's just my own personal preference, but I prefer the `Map.empty` to come first (i.e. the small block of an `if/else`) since it is easier to read.  To find the else is many lines down in the current, but if the smaller if/else block is first then it is easier to find the `else` block.

Reply:
I've inverted it for you ;)